### ISSUE TITLE
Clarify whether the tags `a` and `a=` are the same

### ISSUE THREAD

**Issue by DanielOaks** [ircv3/ircv3-specifications#334]
When parsing tag strings, are `@a=;b` and `@a;b` the same? It feels like software should treat them exactly the same, but making this explicit in the spec would help simplify implementations and make sure clients know they're identical.

**Comment by jwheare** [comment]
From IRC
```
10:32:59 <@jwheare> dan-: fwiw I treat a as a=null and a= as a=‘’
10:33:13 <@dan-> jwheare: I do the same in all my code
```

**Comment by grawity** [comment]
I usually do the same. But, not all languages can distinguish empty string from unset, so IMHO, although clients MAY parse `a` differently from `a=`, tag specifications MUST NOT assign different meanings to them.

**Comment by nomis** [comment]
The specification implies that `a=` is not valid (```<tag> ::= <key> ['=' <escaped value>] ```) because `escaped value` is mandatory.

In charybdis, `a` is a `NULL` value and `a=` is `""`.

**Comment by CarrotCodes** [comment]
Kale does the same as the above - tags are `Map<String, String?>` where `a;b` is `a to null, b to null` and `a=;b` is `a to "", b to null`. Definitely needs clarifying though.

**Comment by DarthGandalf** [comment]
ZNC parses `a` and `a=` the same way, as empty string. When serializing, it writes `=` only if the value is not empty.

**Comment by ilbelkyr** [comment]
@nomis `escaped value` is mandatory, but is defined as

    <sequence of any characters except NUL, CR, LF, semicolon (`;`) and SPACE>

I believe that "sequence of" doesn't necessarily imply "non-empty", but it is perhaps a little unclear.

**Comment by nomis** [comment]
If "sequence of" is not "non-empty" then an empty tag name would be valid.

**Comment by ilbelkyr** [comment]
Good point. It's probably sensible to clarify this, then. (I don't have a strong opinion either way, although @grawity's suggestion appears reasonable.)

**Comment by jwheare** [comment]
#351 got merged but we should check through and see if anything should be added to address this.

**Comment by jwheare** [comment]
This should be resolved by #373
